src/main/java/bio/terra/service/resourcemanagement/ResourceService.java

```diff
@@ -494,6 +500,19 @@ public UUID getOrCreateDatasetProject(
     return googleProjectResource.getId();
   }
 
+  /**
+   * Create a new service account for a dataset to be used to ingest.
+   *
+   * @param projectId the Google id of the project to create the SA for
+   * @param datasetName the name of the dataset the SA is being created for
+   * @return email of the created service account
+   */
+  public String createDatasetServiceAccount(String projectId, String datasetName) {
```

Comment:
With `record` I'm now wondering if we can use it as a way to more narrowly type things that we used to use `String` for.